content-src/asrouter/templates/Trailhead/Trailhead.jsx

```diff
@@ -123,13 +123,18 @@ export class _Trailhead extends React.PureComponent {
     global.addEventListener("visibilitychange", this.closeModal);
   }
 
-  closeModal() {
+  closeModal(ev) {
     global.removeEventListener("visibilitychange", this.closeModal);
     this.props.document.body.classList.remove("welcome");
     this.props.document.getElementById("root").removeAttribute("aria-hidden");
     this.setState({isModalOpen: false});
     this.revealCards();
-    this.props.dispatch(ac.UserEvent({event: "SKIPPED_SIGNIN", ...this._getFormInfo()}));
+
+    // If closeModal() was triggered by a visibilitychange event, the user actually
+    // submitted the email form so we don't send a SKIPPED_SIGNIN ping.
+    if (ev && ev.type !== "visibilitychange") {
```

Comment:
is that too weird/obscure?

Reply:
I guess `!ev || ev.type !== “visibilitychange”` would be better.

Reply:
Yeah !ev should also catch clicking outside / hitting escape. There's 3 callers of closeModal: onsubmit visibilitychange, start button, modaloverlay (keyboard and click)

Reply:
I suppose we could make sure onClose is called always with an event if we fix https://github.com/mozilla/activity-stream/blob/4f311caef9d9aba89126ba26cb1487a7378c5ae6/content-src/asrouter/components/ModalOverlay/ModalOverlay.jsx#L11 to pass in event